data/basics/overflow.go

```diff
@@ -33,6 +33,13 @@ func OAdd16(a uint16, b uint16) (res uint16, overflowed bool) {
 	return
 }
 
+// OAdd32 adds 2 uint16 values with overflow detection
+func OAdd32(a uint32, b uint32) (res uint32, overflowed bool) {
```

Comment:
please add unit tests for these methods.

Reply:
tested AddSaturate32/SubSaturate32

Reply:
comment says it takes uint16s.